### PR TITLE
WIP:TMB-5 - Added Sign up Page

### PR DESCRIPTION
**Pull request by S1ncer3ly**
To Do :-
get otp page

### CODE REVIEW

File: assets/fonts/RozhaOne-Regular.ttf
Comment (A1H1) [inline assets/fonts/RozhaOne-Regular.ttf:1]: make it snake case

File: assets/images/ico_In_flag.png
Comment (A1H1) [inline assets/images/ico_In_flag.png:1]: make it snake case ic_india_flag, also compress it using tinypng

File: lib/app/data/values/images.dart [hunk 1]
```diff
@@ -1,7 +1,9 @@
 class Images {
   Images._privateConstructor();
-
   static const icAppLogo = 'assets/images/ic_app_logo.png';
+  static const icOnboardingBackgroundImage = 'assets/images/bg_onboarding.png';
```
Comment (A1H1) [inline lib/app/data/values/images.dart:4]: its not getting used, its from merge conflict remove it

File: lib/app/data/values/images.dart [hunk 2]
```diff
@@ -1,7 +1,9 @@
...
+  static const icSignUpImage = 'assets/images/ic_email.png';
```
Comment (A1H1) [inline lib/app/data/values/images.dart:5]: wrong variable name, make icEmail

File: lib/app/data/values/images.dart [hunk 3]
```diff
@@ -1,7 +1,9 @@
...
+  static const icInFlag = 'assets/images/ico_In_flag.png';
```
Comment (A1H1) [inline lib/app/data/values/images.dart:6]: ic_india_flag

File: lib/app/modules/auth/signup/views/auth_signup_view.dart [hunk 4]
```diff
@@ -1,18 +1,123 @@
...
 class AuthSignupView extends GetView<AuthSignupController> {
   static launch() => Get.toNamed(Routes.AUTH_SIGNUP);
 
   @override
   Widget build(BuildContext context) {
-    return BaseSignUpView(
-      child: Text("hi"),
+    return Scaffold(
+      body: Padding(
+        padding: const EdgeInsets.fromLTRB(30, 70, 30, 0),
```
Comment (A1H1) [inline lib/app/modules/auth/signup/views/auth_signup_view.dart:18]: there is one dimens file use dimentions from there, check any previous page

File: lib/app/modules/auth/signup/views/auth_signup_view.dart [hunk 5]
```diff
@@ -1,18 +1,123 @@
...
+                    borderColor: AppColors.borderGrey,
+                    iconSize: 40,
+                  ),
+                ),
+                SizedBox(width: 15),
+                Flexible(
+                  fit: FlexFit.tight,
+                  flex: 6,
+                  child: Text(
+                    'What is your mail',
```
Comment (A1H1) [inline lib/app/modules/auth/signup/views/auth_signup_view.dart:37]: hard coded string

File: lib/app/modules/auth/signup/views/auth_signup_view.dart [hunk 6]
```diff
@@ -1,18 +1,123 @@
...
+                    style: TextStyle(fontSize: 32, fontFamily: 'Rozha One'),
```
Comment (A1H1) [inline lib/app/modules/auth/signup/views/auth_signup_view.dart:38]: move to style file

File: lib/app/modules/auth/signup/views/auth_signup_view.dart [hunk 7]
```diff
@@ -1,18 +1,123 @@
...
-      child: Text("hi"),
+    return Scaffold(
+      body: Padding(
+        padding: const EdgeInsets.fromLTRB(30, 70, 30, 0),
+        child: Column(
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            Row(
+              children: [
+                Flexible(
```
Comment (A1H1) [inline lib/app/modules/auth/signup/views/auth_signup_view.dart:24]: flexile is not required here

File: lib/app/modules/auth/signup/views/auth_signup_view.dart [hunk 8]
```diff
@@ -1,18 +1,123 @@
...
+                  flex: 1,
+                  child: CircularContainerWithIcon(
+                    icon: Images.icSignUpImage,
+                    borderColor: AppColors.borderGrey,
+                    iconSize: 40,
+                  ),
+                ),
+                SizedBox(width: 15),
+                Flexible(
```
Comment (A1H1) [inline lib/app/modules/auth/signup/views/auth_signup_view.dart:33]: user expanded

File: lib/app/modules/auth/signup/views/auth_signup_view.dart [hunk 9]
```diff
@@ -1,18 +1,123 @@
...
+                    style: TextStyle(fontSize: 32, fontFamily: 'Rozha One'),
+                  ),
+                ),
+              ],
+            ),
+            SizedBox(height: 30),
+            TextField(
+              decoration: InputDecoration(
+                border: OutlineInputBorder(
+                  borderRadius: BorderRadius.circular(14),
```
Comment (A1H1) [inline lib/app/modules/auth/signup/views/auth_signup_view.dart:47]: move number to dimens

File: lib/app/modules/auth/signup/views/auth_signup_view.dart [hunk 10]
```diff
@@ -1,18 +1,123 @@
...
+                  borderSide: BorderSide(color: AppColors.borderGrey),
+                ),
+                enabledBorder: OutlineInputBorder(
+                  borderRadius: BorderRadius.circular(14),
```
Comment (A1H1) [inline lib/app/modules/auth/signup/views/auth_signup_view.dart:51]: move number to dimens

File: lib/app/modules/auth/signup/views/auth_signup_view.dart [hunk 11]
```diff
@@ -1,18 +1,123 @@
...
+                  borderSide: BorderSide(color: AppColors.borderGrey),
+                ),
+                prefixIcon: Row(
+                  mainAxisSize: MainAxisSize.min,
+                  children: [
+                    Padding(
+                      padding: const EdgeInsets.only(left: 10.0),
```
Comment (A1H1) [inline lib/app/modules/auth/signup/views/auth_signup_view.dart:58]: padding is not required,  use sized box with width

File: lib/app/modules/auth/signup/views/auth_signup_view.dart [hunk 12]
```diff
@@ -1,18 +1,123 @@
...
+                      child: Image.asset(
+                        Images.icInFlag,
+                        width: 24,
```
Comment (A1H1) [inline lib/app/modules/auth/signup/views/auth_signup_view.dart:61]: move to dimnes

File: lib/app/modules/auth/signup/views/auth_signup_view.dart [hunk 13]
```diff
@@ -1,18 +1,123 @@
...
+              decoration: InputDecoration(
+                border: OutlineInputBorder(
+                  borderRadius: BorderRadius.circular(14),
+                  borderSide: BorderSide(color: AppColors.borderGrey),
+                ),
+                enabledBorder: OutlineInputBorder(
+                  borderRadius: BorderRadius.circular(14),
+                  borderSide: BorderSide(color: AppColors.borderGrey),
+                ),
+                labelText: ' Email Id *',
```
Comment (A1H1) [inline lib/app/modules/auth/signup/views/auth_signup_view.dart:93]: move to strings

File: lib/app/modules/auth/signup/views/auth_signup_view.dart [hunk 14]
```diff
@@ -1,18 +1,123 @@
...
+                floatingLabelBehavior: FloatingLabelBehavior.always,
+                contentPadding:
+                    EdgeInsets.symmetric(horizontal: 27, vertical: 18),
```
Comment (A1H1) [inline lib/app/modules/auth/signup/views/auth_signup_view.dart:96]: move to dimens and use even numbers

File: lib/app/modules/auth/signup/views/auth_signup_view.dart [hunk 15]
```diff
@@ -1,18 +1,123 @@
...
+              ),
+            ),
+            SizedBox(
+              height: 23,
```
Comment (A1H1) [inline lib/app/modules/auth/signup/views/auth_signup_view.dart:100]: move to dimens and use even numbers

File: lib/app/modules/auth/signup/views/auth_signup_view.dart [hunk 16]
```diff
@@ -1,18 +1,123 @@
...
+            ),
+            Text(
+              Strings.signUpPageInfo,
+              style: TextStyle(
```
Comment (A1H1) [inline lib/app/modules/auth/signup/views/auth_signup_view.dart:104]: move to styles

File: lib/app/modules/auth/signup/views/auth_signup_view.dart [hunk 17]
```diff
@@ -1,18 +1,123 @@
...
+                  color: AppColors.lightGrey,
+                  fontWeight: FontWeight.w100,
+                  fontSize: 12),
+            ),
+            SizedBox(
+              height: 80,
+            ),
+            FilledRoundedButton(
+              width: 203,
```
Comment (A1H1) [inline lib/app/modules/auth/signup/views/auth_signup_view.dart:113]: remove width , and avoid hardcoded wdith, use padding instead

File: lib/widgets/buttons/filled_round_button.dart [hunk 18]
```diff
@@ -7,11 +7,13 @@ class FilledRoundedButton extends StatelessWidget {
   final String text;
   final VoidCallback onTap;
   final Color backgroundColor;
+  final double width; // Custom width
```
Comment (A1H1) [inline lib/widgets/buttons/filled_round_button.dart:10]: remove this

File: lib/widgets/circular_container_with_icon/circular_container_with_icon.dart [hunk 19]
```diff
@@ -0,0 +1,48 @@
+import 'package:flutter/material.dart';
+import 'package:starter/app/theme/app_colors.dart';
+
+class CircularContainerWithIcon extends StatelessWidget {
+  final Color backgroundColor;
+  final Color borderColor;
+  final dynamic icon;
```
Comment (A1H1) [inline lib/widgets/circular_container_with_icon/circular_container_with_icon.dart:7]: data type  should not be dynamic

**Review by A1H1 — CHANGES_REQUESTED** [review]

File: lib/app/data/values/images.dart [hunk 20]
```diff
@@ -1,7 +1,8 @@
 class Images {
   Images._privateConstructor();
-
   static const icAppLogo = 'assets/images/ic_app_logo.png';
+  static const icEmail = 'assets/images/ic_email.png';
+  static const ic_in_flag = 'assets/images/ico_In_flag.png';
```
Comment (istutiagarwal) [inline lib/app/data/values/images.dart:5]: please change the name to 'ic_flag_india'  for file name and icFlagIndia for variable name.

File: lib/app/modules/auth/signup/views/auth_signup_view.dart [hunk 21]
```diff
@@ -1,18 +1,124 @@
...
+            ),
+            SizedBox(
+              height:Dimens.dimens_24,
+            ),
+            Text(
+              Strings.signUpPageInfo,
+              style: Styles.tsLightGrey12,
+            ),
+            SizedBox(
+              height: 80,
```
Comment (istutiagarwal) [inline lib/app/modules/auth/signup/views/auth_signup_view.dart:109]: here again

File: lib/widgets/circular_container_with_icon/circular_container_with_image.dart [hunk 22]
```diff
@@ -0,0 +1,41 @@
...
+  final double borderWidth;
+
+  const CircularContainerWithImage({
+    Key? key,
+    this.backgroundColor = AppColors.white,
+    required this.icon,
+    this.iconColor = AppColors.white,
+    this.iconSize = 14,
+    this.borderColor = Colors.black,
+    this.borderWidth = 1.0,
```
Comment (istutiagarwal) [inline lib/widgets/circular_container_with_icon/circular_container_with_image.dart:19]: here also

File: lib/widgets/circular_container_with_icon/circular_container_with_image.dart [hunk 23]
```diff
@@ -0,0 +1,41 @@
...
+  final double iconSize;
+  final Color borderColor;
+  final double borderWidth;
+
+  const CircularContainerWithImage({
+    Key? key,
+    this.backgroundColor = AppColors.white,
+    required this.icon,
+    this.iconColor = AppColors.white,
+    this.iconSize = 14,
```
Comment (istutiagarwal) [inline lib/widgets/circular_container_with_icon/circular_container_with_image.dart:17]: here also

File: lib/widgets/circular_container_with_icon/circular_container_with_image.dart [hunk 24]
```diff
@@ -0,0 +1,41 @@
...
+    return Container(
+      decoration: BoxDecoration(
+        color: backgroundColor,
+        shape: BoxShape.circle,
+        border: Border.all(
+          color: borderColor,
+          width: borderWidth,
+        ),
+      ),
+      padding: EdgeInsets.all(4),
```
Comment (istutiagarwal) [inline lib/widgets/circular_container_with_icon/circular_container_with_image.dart:33]: here also

**Pull request merged**